lamindb/_curate.py

```diff
@@ -29,8 +29,8 @@ class CurateLookup:
 
     def __init__(
         self,
-        categoricals: dict[str, FieldAttr],
-        slots: dict[str, FieldAttr] = None,
+        categoricals: Mapping[str, FieldAttr],
```

Comment:
See below: https://github.com/laminlabs/lamindb/pull/1820/files#r1719822354

Reply:
I think the question is if we want `categoricals` to be something else than a Python dict. Do we want to allow a `pd.Series`, for instance?

Reply:
A `pd.Series` not necessarily but `OrderedDict` and friends.

Reply:
`pd.Series` is not of type `typing.Mapping` I think. It's `Iterable`

Reply:
`OrderedDict` has looooong been a legacy concept, since 3.6 already?

And even if we'd want to account for it: isn't it a subclass of `dict` and hence encompassed in the type annotation?

I'd use `Mapping` if I'd actually want to convey to the user that they can pass something like `pd.Series`.

Reply:
> pd.Series is not of type typing.Mapping I think. It's Iterable

In my view of the world `pd.Series` is both `Mapping` and `Iterable` in a more subtle way (by virtue of being key-ed) than `Mapping` itself is an `Iterable`.

Reply:
No, `OrderedDict` is a separate class in `Collection`. Another candidate would be `defaultdict` but this one actually is a subtype of `dict` I believe.

> I'd use Mapping if I'd actually want to convey to the user that they can pass something like pd.Series.

pd.Series is NOT of type Mapping AFAIK.

It's NBD, I can change it back to dict if you prefer. Doesn't make big difference.

Reply:
`OrderedCollection` is a subclass of dict:
<img width="798" alt="image" src="https://github.com/user-attachments/assets/0b3d94e5-034c-49c6-802b-224d5d599127">

https://docs.python.org/3/library/collections.html#collections.OrderedDict

What I didn't know is that there still seems to be a reason people might use it. I stopped using it when 3.6 came out and `dict` started to be ordered.

Reply:
> In my view of the world pd.Series is both Mapping and Iterable in a more subtle way (by virtue of being key-ed) than Mapping itself is an Iterable.

This would only confuse type checkers and users. The classes/implementations define what they are and we should stick to that.

Reply:
> pd.Series is NOT of type Mapping AFAIK.

Indeed. It behaves like my common-sense understanding of a "mapping" but it's not `typing.Mapping` which refers to the `collections.abc` `Mapping`... that's confusing to me...

Asked Claude:
<img width="695" alt="image" src="https://github.com/user-attachments/assets/1ec5dec5-f67e-43a1-88d3-b351c2dda835">

Here is the discussion: https://claude.ai/share/e01aca91-880b-417d-90d0-92daefeb068d

I'm leaning towards `dict` for the reasons mentioned but I guess `Mapping` can also be argued for even though I don't see what people might want to pass that's not a subclass of `dict` (other than `pd.Series` which isn't covered by `Mapping`).